frontend/tests/dangerZone.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { Page, expect } from "@playwright/test";
+import { test, realms } from "./util/common";
+import { USERS, login } from "./util/user";
+
+for (const realmType of realms) {
+    test(`${realmType} danger zone`, async ({
+        page, browserName, standardData, activeSearchIndex,
+    }) => {
+        test.skip(browserName === "webkit", "Skip safari because it doesn't allow http logins");
+
+        const userid = realmType === "User" ? "morgan" : "sabine";
+        const parentPageName = realmType === "User" ? USERS[userid] : "Far side";
+        await test.step("Setup", async () => {
+            await page.goto("/");
+            await login(page, userid);
+
+            // Go to a non-root realm
+            if (realmType === "Regular") {
+                await page.locator("nav").getByRole("link", { name: parentPageName }).click();
+                await expect(page).toHaveURL("/moon");
+            }
+
+            // Create user realm
+            if (realmType === "User") {
+                await page.goto(`/@${userid}`);
+                await expect(page).toHaveURL(`/@${userid}`);
+            }
+
+            await page.getByRole("link", { name: "Page settings" }).click();
+        });
+
+        if (realmType === "Regular") {
+            await test.step("Path changing", async () => {
+                await test.step("Path can be changed", async () => {
+                    const pathInput = page.locator("input[name='pathSegment']");
+                    await pathInput.fill("new-path");
+                    await page.getByRole("button", { name: "Change path segment" }).click();
+
+                    await expect(page).toHaveURL("~manage/realm?path=%2Fnew-path");
+                });
+
+                await test.step("Links are updated", async () => {
+                    const links = [
+                        ["Go to page", "Far side"],
+                        ["Page settings", "Settings of page “Far side”"],
+                        ["Edit page content", "Edit page “Far side”"],
+                        ["Add subpage", "Add page"],
+                        ["Finanzamt", "Finanzamt"],
+                        ["Far side", "Far side"],
+                    ];
+
+                    const linkTest = async (page: Page, linkName: string, heading: string) => {
+                        await page.getByRole("link", { name: linkName }).first().click();
+                        await expect(page.getByRole("heading", { name: heading })).toBeVisible();
+                    };
+
+                    for (const [name, heading] of links) {
+                        await linkTest(page, name, heading);
+                    }
```

Comment:
Why not just inline this function? Like... it's only called once. You might not have had a loop before, making it useful to extract it into a function, but like this I don't think it serves a purpose.

I'm also fine with not having a loop and instead calling `linkTest` a bunch of times. Might be nicer as somehow I found it a bit surprising that the loop over that data has side effects (it goes to different pages). Also, the `linkTest` function could just capture `page` right? It doesn't need to be passed as argument. Finally, `testLink` might be a better name?